src/sentry/static/sentry/app/api.tsx

```diff
@@ -266,20 +334,35 @@ export class Client {
               path,
               requestOptions: options,
             },
-            ...args
+            resp,
+            textStatus,
+            errorThrown
           );
         },
-        complete: (...args) => {
+        complete: (jqXHR: JQueryXHR, textStatus: string) => {
           Sentry.finishSpan(requestSpan);
-          return this.wrapCallback(id, options.complete, true)(...args);
+          return this.wrapCallback<[JQueryXHR, string]>(id, options.complete, true)(
+            jqXHR,
+            textStatus
+          );
         },
       })
     );
 
     return this.activeRequests[id];
   }
 
-  requestPromise(path, {includeAllArgs, ...options} = {}) {
+  requestPromise<IncludeAllArgsType extends boolean>(
+    path: string,
+    {
+      includeAllArgs,
+      ...options
+    }: {includeAllArgs?: IncludeAllArgsType} & Readonly<RequestOptions> = {}
+  ): Promise<
+    IncludeAllArgsType extends true
+      ? [any, string | undefined, JQueryXHR | undefined]
+      : any
+  > {
```

Comment:
@markstory I'm able to add a conditional type so that the right promise type is returned depending on if `includeAllArgs` is `true`.